cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs

```diff
@@ -0,0 +1,266 @@
+-- wait for the number of DReps being @n@ for two epochs. If
+-- this number is not attained before two epochs, the test is failed.
+waitDRepsNumber ::
+  (HasCallStack, MonadIO m, MonadCatch m, MonadTest m)
+  => NodeConfigFile 'In
+  -> SocketPath
+  -> H.ExecConfig
+  -> Int
+  -> m ()
+waitDRepsNumber configurationFile socketPath execConfig expectedDRepsNb = do
+  QueryTipLocalStateOutput{mEpoch} <- P.execCliStdoutToJson execConfig [ "query", "tip" ]
+  currentEpoch <- H.evalMaybe mEpoch
+  let terminationEpoch = succ . succ $ currentEpoch
+  void $ H.evalMaybeM $ waitDRepsNumber' configurationFile socketPath terminationEpoch expectedDRepsNb
+
+-- | @waitDRepsNumber' config socket terminationEpoch n@
+-- wait until @terminationEpoch@ for the number of DReps being @n@. If
+-- this number is not attained before @terminationEpoch@, the test is failed.
+-- So if you call this function, you are expecting the number of DReps to already
+-- be @n@, or to be @n@ before @terminationEpoch@
+waitDRepsNumber' ::
+  (HasCallStack, MonadIO m, MonadTest m)
+  => NodeConfigFile In
+  -> SocketPath
+  -> EpochNo -- ^ The termination epoch: the constitution proposal must be found *before* this epoch
+  -> Int -- ^ The expected numbers of DReps. If this number is not reached until the termination epoch, this function fails the test.
+  -> m (Maybe [L.DRepState StandardCrypto]) -- ^ The DReps when the expected number of DReps was attained.
+waitDRepsNumber' nodeConfigFile socketPath maxEpoch expectedDRepsNb = do
+  result <- runExceptT $ checkLedgerStateCondition nodeConfigFile socketPath QuickValidation maxEpoch Nothing
+      $ \(AnyNewEpochState actualEra newEpochState) -> do
+        case testEquality expectedEra actualEra of
+          Just Refl -> do
+            let dreps = Map.elems $ shelleyBasedEraConstraints expectedEra newEpochState
+                      ^. L.nesEsL
+                      . L.esLStateL
+                      . L.lsCertStateL
+                      . L.certVStateL
+                      . L.vsDRepsL
+            if length dreps == expectedDRepsNb then do
+                put $ Just dreps
+                pure ConditionMet
+            else
+                pure ConditionNotMet
+          Nothing -> do
+            error $ "Eras mismatch! expected: " <> show expectedEra <> ", actual: " <> show actualEra
+  case result of
+    Left (FoldBlocksApplyBlockError (TerminationEpochReached epochNo)) -> do
+      H.note_ $ "waitDRepsNumber: drep number did not become " <> show expectedDRepsNb <> " before termination epoch: " <> show epochNo
+      H.note_ "This is likely an error of this test."
+      H.failure
+    Left err -> do
+      H.note_ $ "waitDRepsNumber: could not reach termination epoch: " <> docToString (prettyError err)
+      H.note_ "This is probably an error unrelated to this test."
+      H.failure
+    Right (_, val) ->
+      return val
+  where
+    expectedEra = ShelleyBasedEraConway
```

Comment:
You have defined the era used in this test twice. Here and in line 60.

Reply:
Made the era a top-level constant :+1: